Added Term Entry HTML: Tags: `<source>` (#560)
Codecademy/docs: content/html/concepts/tags/terms/source/source.md

---
Title: '<source>'
Description: 'Represents an interface for adding source content to the page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:

>   - 'Tags'

Reviewer: ```suggestion
  - 'Tags'
  - 'DOM'
  - 'Elements'
  - 'Semantic'
```

CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

> The `<source>` element allows for multiple sources to be in one element in case one type is not supported. The `<source>` element is nested within an `<video>`, `<img>` or another tag that embeds content. It allows for multiple files to be in one element in case one file type is not supported the video can still play with a different type.  Here's an example:

Reviewer: ```suggestion
The `<source>` element allows for multiple sources to be in one element in case one type is not supported. The `<source>` element is nested within an `<video>`, `<img>` or another tag that embeds content. It allows for multiple files to be referenced in one element in case a file type is not supported. If one file type can not be displayed, the browser will attempt to display the next.
```

> 

Reviewer: ```suggestion

## Example

```

```html

> 

Reviewer: ```suggestion
```

<video>
  <source src = "./video.mp4" type="video/mp4">
  <source src = "./video.webm" type="video/webm">
  Video not supported
</video>
```

> The browser would first try and play the `.mp4` video and if it can't it will try the `.webm` video. And if neither of those work it will display `Video not supported`. You can have as many `<source>` elements as you want as long as you have enough files for each on

Reviewer: ```suggestion
The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. And if neither of those work it will display "Video not supported". You can have as many `<source>` elements as you want as long as you have files for each one.
```

---
Title: '<source>'
Description: 'Represents an interface for adding source content to the page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Tags'
  - 'DOM'
  - 'Elements'
CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

> The `<source>` element allows for multiple sources to be in one element in case one type is not supported. The `<source>` element is nested within an `<video>`, `<img>` or another tag that embeds content. It allows for multiple files to be referenced in one element in case a file type is not supported. If one file type can not be displayed, the browser will attempt to display the next.

Reviewer 2: ```suggestion
The `<source>` element allows for multiple sources or file formats to be associated with the `<audio>`, `<video>`, and `<picture>` tags. In case one source is incompatible with a given browser, another one can be referenced. If all else fails, the browser will attempt to display fallback content.
```

Reviewer 2: There's no compatibility between the `<img>` and `<source>` elements, AFAIK. The `<img>` tag has a `src` attribute, after all. 😄

> 

Reviewer 2: @ethan-master-coding Let's make sure to add a brief "Syntax" section here above the "Example". This is a format that we'd like to follow for term entries.

## Example

```html

> <video>

Reviewer 2: ```suggestion
<video controls>
```

This will allow the empty video screen to be visible on most browsers.

  <source src="./video.mp4" type="video/mp4" />
  <source src="./video.webm" type="video/webm" />
  Video not supported
</video>
```

> The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. And if neither of those work it will display "Video not supported". A large amount of `<source>` elements can be used as long as there is a valid file sourced for each one.

Reviewer 2: ```suggestion
The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. If neither of those work one of the following will happen:

* `<video>` will remain inline within the page's markup with only the controls visible (if they're enabled).
* Fallback content (like "Video not supported") is displayed for browsers that do not support HTML5, which includes the `<audio>` or `<video>` tags.

A large amount of `<source>` elements can be used as long as there is a valid file sourced for each one.
```

---
Title: '<source>'
Description: 'Represents an interface for adding source content to the page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Tags'
  - 'DOM'
  - 'Elements'
  - 'Semantic'
CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

The `<source>` element allows for multiple sources to be in one element in case one type is not supported. The `<source>` element is nested within an `<video>`, `<img>` or another tag that embeds content. It allows for multiple files to be referenced in one element in case a file type is not supported. If one file type can not be displayed, the browser will attempt to display the next.

## Example

```html
<video>
  <source src = "./video.mp4" type="video/mp4">
  <source src = "./video.webm" type="video/webm">
  Video not supported
</video>
```

> The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. And if neither of those work it will display "Video not supported". You can have as many `<source>` elements as you want as long as you have files for each one.

Reviewer 2: ```suggestion
The browser will first try and play the **./video.mp4** file and if it can't it will try the **./video.webm** file. And if neither of those work it will display "Video not supported".  A large amount of `<source>` elements can be used as long as there is a valid file sourced for each one.
```

For Docs entries, we want to avoid using 1st- or 2nd-person pronouns and opt for a friendly, but manual voice that uses 3rd-person sparingly.

---
Title: '<source>'
Description: 'Represents an interface for adding source content to the page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Tags'
  - 'DOM'
  - 'Elements'
CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

The `<source>` element allows for multiple sources or file formats to be associated with the `<audio>`, `<video>`, and `<picture>` tags. In case one source is incompatible with a given browser, another one can be referenced. If all else fails, the browser will attempt to display fallback content. The `<sourse>` tag is a child of a `<audio>`, `<video>`, or `<picture>` and has atleast 1 attribute. The `src` attribute needs to be in every `<source>` tag with the file path to the media.

> 

Reviewer 3: Add backticks to end the code block. I can't do this in the comment because it interferes with the suggestion block.

> ## Example

Reviewer 3: ```suggestion

Attributes provide additional characteristics to the element. Some common attributes used with `<source>` are:
  - `src`: specifies the URL of the media, used for `<audio>` and `<video>` tags
  - `srcset`: defines a set of images for the browser to choose from, used with `<picture>`  
  - `media`: tells which device the document is optimized for
  - `sizes`: sets out the size of visual media icons
  - `type`: specifies the content type e.g., `video/mp4` or `audio/ogg`
	  
## Example
```

---
Title: '<source>'
Description: 'Represents an interface for adding source content to the page.'
Subjects:
  - 'Web Development'
  - 'Web Design'
Tags:
  - 'Tags'
  - 'DOM'
  - 'Elements'
CatalogContent:
  - 'learn-html'
  - 'paths/front-end-engineer-career-path'
---

The `<source>` element allows for multiple sources or file formats to be associated with the [`<audio>`](https://www.codecademy.com/resources/docs/html/tags/audio), [`<video>`](https://www.codecademy.com/resources/docs/html/tags/video), and `<picture>` tags. In case one source is incompatible with a given browser, another one can be referenced. If all else fails, the browser will attempt to display fallback content. The `<source>` tag is a child of a `<audio>`, `<video>`, or `<picture>` and has at least one [attribute](https://www.codecademy.com/resources/docs/html/attributes). Note that with the `<picture>` tag the `srcset` is perfered over `src`.

## Syntax

> `<source>` is an empty element. It has no child nodes and no closing tag.

Reviewer 3: ```suggestion
`<source>` is an empty tag and has no children or closing tags.
```

```pseudo 
       <source <!-- attributes go here --> > 
```

Attributes provide additional characteristics to the element. Some common attributes used with `<source>` are:
  - `src`: specifies the URL of the media, used for `<audio>` and `<video>` tags
  - `srcset`: defines a set of images for the browser to choose from, used with `<picture>`  
  - `media`: tells which device the document is optimized for
  - `sizes`: sets out the size of visual media icons

>   - `type`: specifies the content type e.g., `video/mp4` or `audio/ogg`

Reviewer 3: ```suggestion
  - `type`: specifies the content type e.g., `video/mp4` or `audio/ogg`, can be used with any of the three parent elements of `<source>` (`<picture>`, `<video>`, or `<audio>`).
```
